ts/components/conversation/SessionConversation.tsx

```diff
@@ -221,7 +228,39 @@ export class SessionConversation extends React.Component<Props, State> {
       return <MessageView />;
     }
 
+    const conversation = getConversationController().get(selectedConversation.id);
+    const isApproved = conversation.isApproved();
     const selectionMode = selectedMessages.length > 0;
+    const useMsgRequests =
+      window.lokiFeatureFlags.useMessageRequests &&
+      window.inboxStore?.getState().userConfig.messageRequests;
+    const showMsgRequestUI = useMsgRequests && !isApproved && messagesProps.length > 0;
+
+    const handleDeclineConversationRequest = async () => {
+      window.inboxStore?.dispatch(
+        updateConfirmModal({
+          okText: window.i18n('decline'),
+          cancelText: window.i18n('cancel'),
+          message: window.i18n('declineRequestMessage'),
+          onClickOk: async () => {
+            declineConversation(selectedConversation.id, false);
+            blockConvoById(selectedConversation.id);
+            forceSyncConfigurationNowIfNeeded();
+          },
```

Comment:
I think you should await each of these calls here, as the last one might not take an uptodate value to do the sync otherwise.